pkg/controller/account/account_controller.go

```diff
@@ -631,6 +625,81 @@ func (r *ReconcileAccount) statusUpdate(reqLogger logr.Logger, account *awsv1alp
 	return err
 }
 
+func (r *ReconcileAccount) setStateFailed(reqLogger logr.Logger, currentAcctInstance *awsv1alpha1.Account, msg string) error {
+	reqLogger.Info(msg)
+	// Upodate the account status (state and condition) to Failed
+	utils.SetAccountStatus(reqLogger, currentAcctInstance, msg, v1alpha1.AccountFailed, AccountFailed)
+	err := r.statusUpdate(reqLogger, currentAcctInstance)
+	if err != nil {
+		return err
+	}
+
+	// Set a failure condition in the accountClaim
+	err = r.setAccountClaimError(reqLogger, currentAcctInstance, msg)
+	if err != nil {
+		return err
+	}
+
+	// Apply the update
+	err = r.statusUpdate(reqLogger, currentAcctInstance)
+	return err
+}
+
+func (r *ReconcileAccount) getAccountClaim(account *awsv1alpha1.Account) (*awsv1alpha1.AccountClaim, error) {
+	accountClaim := &awsv1alpha1.AccountClaim{}
+	err := r.Client.Get(context.TODO(), types.NamespacedName{
+		Name: account.Spec.ClaimLink, Namespace: account.Spec.ClaimLinkNamespace}, accountClaim)
+	if err != nil {
+		return nil, err
+	}
+	return accountClaim, nil
+}
+
+func (r *ReconcileAccount) setAccountClaimError(reqLogger logr.Logger, currentAccountInstance *awsv1alpha1.Account, message string) error {
```

Comment:
As noted above, IMO this should just go away. But if you were going to keep it, why wouldn't it call the corresponding func from the accountclaim package? Circular import? (Which may be another hint that we shouldn't be updating the same object from both places.)

Reply:
Yeah, I was struggling with circular imports.  I will flag this for consideration in a future PR to address across-controller resource handling.